fixtures/soft_assert.py

```diff
@@ -89,8 +100,8 @@ def _soft_assert_cm():
     """
     _thread_locals.caught_asserts = []
     yield _thread_locals.caught_asserts
-    if _thread_locals.caught_asserts:
-        raise SoftAssertionError(_thread_locals.caught_asserts)
+    # if _thread_locals.caught_asserts:
+    #     raise SoftAssertionError(_thread_locals.caught_asserts)
```

Comment:
this context manager is used only inside soft_assert and works right only for cases when soft_assert is used inside the test. It can't work when it is used in other fixtures.
So, I moved this raise statement to teardown step as this was only good way how to handle both cases.